directory/datadog_checks/directory/directory.py

```diff
@@ -32,81 +33,40 @@ class DirectoryCheck(AgentCheck):
-        recursive = is_affirmative(instance.get('recursive', False))
-        dirtagname = instance.get('dirtagname', 'name')
-        filetagname = instance.get('filetagname', 'filename')
-        filegauges = is_affirmative(instance.get('filegauges', False))
-        countonly = is_affirmative(instance.get('countonly', False))
-        ignore_missing = is_affirmative(instance.get('ignore_missing', False))
-        follow_symlinks = is_affirmative(instance.get('follow_symlinks', True))
-        custom_tags = instance.get('tags', [])
-        max_filegauge_count = instance.get('max_filegauge_count', self.MAX_FILEGAUGE_COUNT)
-
-        if not exists(abs_directory):
+
+    def __init__(self, *args, **kwargs):
+        # type: (*Any, **Any) -> None
+        super(DirectoryCheck, self).__init__(*args, **kwargs)
+
+        self.config = DirectoryConfig(self.instance)
+
+    def check(self, _):
+        if not exists(self.config.abs_directory):
             msg = (
                 "Either directory '{}' doesn't exist or the Agent doesn't "
-                "have permissions to access it, skipping.".format(abs_directory)
+                "have permissions to access it, skipping.".format(self.config.abs_directory)
             )
 
-            if not ignore_missing:
+            if not self.config.ignore_missing:
                 raise ConfigurationError(msg)
 
             self.log.warning(msg)
 
-        self._get_stats(
-            abs_directory,
-            name,
-            dirtagname,
-            filetagname,
-            filegauges,
-            pattern,
-            exclude_dirs_pattern,
-            dirs_patterns_full,
-            recursive,
-            follow_symlinks,
-            countonly,
-            custom_tags,
-            max_filegauge_count,
-        )
+        self._get_stats()
 
-    def _get_stats(
-        self,
-        directory,
-        name,
-        dirtagname,
-        filetagname,
-        filegauges,
-        pattern,
-        exclude_dirs_pattern,
-        dirs_patterns_full,
-        recursive,
-        follow_symlinks,
-        countonly,
-        tags,
-        max_filegauge_count,
-    ):
-        dirtags = ['{}:{}'.format(dirtagname, name)]
-        dirtags.extend(tags)
+    def _get_stats(self):
+        dirtags = ['{}:{}'.format(self.config.dirtagname, self.config.name)]
+        dirtags.extend(self.config.tags)
         directory_bytes = 0
         directory_files = 0
-        max_filegauge_balance = max_filegauge_count
+        max_filegauge_balance = self.config.max_filegauge_count
 
         # If we do not want to recursively search sub-directories only get the root.
-        walker = walk(directory, follow_symlinks) if recursive else (next(walk(directory, follow_symlinks)),)
+        walker = (
+            walk(self.config.abs_directory, self.config.follow_symlinks)
+            if self.config.recursive
+            else (next(walk(self.config.abs_directory, self.config.follow_symlinks)),)
```

Comment:
(Nit) The usage of `next()` is a bit surprising at first, but now I get it and I'm not sure how else I'd write this. Tho maybe like this…

```python
walker = walk(self.config.abs_directory, self.config.follow_symlinks)
if not self.config.recursive:
    # Only visit the first directory.
    walker = [next(walker)]
```

Reply:
Yes, though about changing that part too, but wanted to scope this PR to only the config part.

Reply:
@florimondmanca https://github.com/DataDog/integrations-core/pull/6965